### PR TITLE
GlyphPathBuilder: support contours starting with offcurves or without oncurves

### LINKED ISSUE #349: Google Sans has outlines we don't like
https://github.com/googlefonts/googlesans/issues/586

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes #349

### CODE REVIEW

File: glyphs2fontir/src/toir.rs [hunk 1]
```diff
@@ -49,34 +49,27 @@ fn to_ir_path(glyph_name: GlyphName, src_path: &Path) -> Result<BezPath, WorkErr
     // See also https://github.com/fonttools/ufoLib2/blob/4d8a9600148b670b0840120658d9aab0b38a9465/src/ufoLib2/pointPens/glyphPointPen.py#L16
     let mut path_builder = GlyphPathBuilder::new(glyph_name.clone());
     if src_path.nodes.is_empty() {
-        return Ok(path_builder.build());
+        return Ok(path_builder.build()?);
     }
 
-    let mut nodes = &src_path.nodes[..];
+    let mut nodes: Vec<_> = src_path.nodes.iter().collect();
```
Comment (cmyr) [inline glyphs2fontir/src/toir.rs:55]: This could just be `src_path.nodes.clone()`?

**Review by cmyr — APPROVED** [review]
Cool, this looks good to me, and the tests/impl ratio is encouraging. A few little notes inline, but nothing major :)

File: glyphs2fontir/src/toir.rs (hunk 1 above)
Comment (anthrotype) [reply glyphs2fontir/src/toir.rs]: Done. Why didn't clippy suggest that?!

File: fontir/src/ir.rs [hunk 2]
```diff
@@ -1028,33 +1058,58 @@ impl GlyphPathBuilder {
...
+    fn begin_path(&mut self, oncurve: OnCurve) -> Result<(), PathConversionError> {
+        assert!(self.first_oncurve.is_none());
+        self.path.move_to(*oncurve.point());
+        self.first_oncurve = Some(oncurve);
         Ok(())
     }
 
+    /// Lifts the "pen" to Point `p` and marks the beginning of an open contour.
+    /// A point of this type can only be the first point in a contour.
+    /// Cf. "move" in https://unifiedfontobject.org/versions/ufo3/glyphs/glif/#point-types
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply fontir/src/ir.rs]: Changed it so that first line is short summary, then empty line, then more details if any. Wrapped direct links in `<>`

**Comment by anthrotype** [comment]
I'd like to merge this before it gets out of sync, I can address any review commens e.g. Rod may have once he gets back

**Pull request merged**